Demo/UI Tests/Helpers/BTUITest.swift

```diff
@@ -1,32 +1,52 @@
 import XCTest
 
 extension XCTestCase {
-
-    func waitForElementToAppear(_ element: XCUIElement, timeout: TimeInterval = 30) {
+    
+    /// Wait for element to appear
+    @discardableResult
+    func waitForElementToAppear(_ element: XCUIElement, timeout: TimeInterval = 30) -> Bool {
         let existsPredicate = NSPredicate(format: "exists == true")
-        
-        expectation(for: existsPredicate, evaluatedWith: element)
-        
-        waitForExpectations(timeout: timeout)
+        let expectation = expectation(for: existsPredicate, evaluatedWith: element)
+        let result = XCTWaiter.wait(for: [expectation], timeout: timeout)
+        return result == .completed
     }
     
-    func waitForElementToBeHittable(_ element: XCUIElement, timeout: TimeInterval = 30) {
-        let existsPredicate = NSPredicate(format: "exists == true && hittable == true")
-        
-        expectation(for: existsPredicate, evaluatedWith: element)
-        
-        waitForExpectations(timeout: timeout)
+    /// Wait for element to be hittable
+    @discardableResult
+    func waitForElementToBeHittable(_ element: XCUIElement, timeout: TimeInterval = 30) -> Bool {
+        let predicate = NSPredicate(format: "exists == true && hittable == true && enabled == true")
+        let expectation = expectation(for: predicate, evaluatedWith: element)
+        let result = XCTWaiter.wait(for: [expectation], timeout: timeout)
+        return result == .completed
     }
 }
 
+// MARK: - XCUIElement Extensions
+
 extension XCUIElement {
 
-    func forceTapElement() {
-        if self.isHittable {
-            self.tap()
-        } else {
-            let coordinate: XCUICoordinate = self.coordinate(withNormalizedOffset: CGVector(dx: 0.0, dy: 0.0))
-            coordinate.tap()
+    /// Improved tap with retry logic
```

Comment:
Should we remove these comments? We may not need to reference the previous state.